beamer/deploy/util.py

```diff
@@ -0,0 +1,143 @@
+import json
+import subprocess
+from pathlib import Path
+from typing import Any, Sequence, Union, cast
+
+import structlog
+from eth_typing import ChecksumAddress
+from eth_utils import encode_hex, to_wei
+from web3 import Web3
+from web3.contract import Contract, ContractConstructor
+from web3.contract.contract import ContractFunction
+
+import beamer.deploy.config as config
+from beamer.util import transact
+
+log = structlog.get_logger(__name__)
+
+
+class DeployedContract(Contract):
+    deployment_block: int
+    deployment_txhash: str
+    deployment_args: list[Any]
+    name: str
+
+
+def _transact(func: Union[ContractConstructor, ContractFunction]) -> Any:
+    return transact(func, timeout=600, poll_latency=1)
+
+
+def _load_contracts_info(contracts_path: Path) -> dict[str, tuple]:
```

Comment:
maybe, it is better to use built_path or something here because we would be using contracts/.build which has ape abi and __local__.json